core/src/main/java/org/apache/iceberg/rest/ResourcePaths.java

```diff
@@ -18,17 +18,30 @@
  */
 package org.apache.iceberg.rest;
 
+import java.util.Locale;
 import java.util.Map;
 import org.apache.iceberg.catalog.Namespace;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 
 public class ResourcePaths {
   private static final Joiner SLASH = Joiner.on("/").skipNulls();
   private static final String PREFIX = "prefix";
+  private static final String SNAPSHOT_LOADING_BEHAVIOR = "snapshot-loading";
```

Comment:
Why was this added to `ResourcePaths`? It seems like a separate concern and it's confusing when I see calls like `paths.snapshotLoading()`. Why not just handle params to the table load path separately?

This is also a public class and if we intend to change how this works (by adding a snapshots route later) then we'll need to unwind this.

Reply:
I put this in `ResourcePaths` because paths and params are related and all part of the request URI.  We can't build the full uri for a request because the api for the `RESTClient` separates path and params.  These are the params to the load table path, so maybe it would be more clear if it were `paths.tableParams()`? (this would correspond to the existing `params.table()`

Reply:
I agree that it's confusing seeing `paths.snapshotLoading()` calls in `RESTSessionCatalog`